### PR TITLE
Add AutoPayService for automatic bill processing

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context

**Type:** Feat

**Intent:** Implement an automated bill payment processing system that periodically processes bills marked for automatic payment, creating transaction records and advancing recurring bills to their next due dates while marking one-time bills as paid. This introduces a new scheduled job and service to handle recurring billing workflows with resilience to per-bill failures.

### 🧭 Reviewer Guide

**Entry Point:** Start with `lib/services/AutoPayService.ts` (135 lines) to understand the core business logic - eligibility criteria (isAutoPay = true, status != 'paid', dueDate <= end of today, not archived), transaction creation, and bill advancement via RecurrenceService. Then examine `lib/services/SchedulerService.ts` to see how the new auto-pay-processor job (daily at 00:05) is integrated into the existing scheduler, and review the comprehensive test suite to validate expected behaviors.

**Complexity:** Medium

**Estimated Review Time:** ~40 minutes

**Sensitive Areas:**
- `lib/services/AutoPayService.ts` - Database transaction handling and state mutations for billing records; failure resilience patterns; correct integration with RecurrenceService for advancing due dates
- `lib/services/AutoPayService.test.ts` (496 lines) - Comprehensive mocking strategy for database and RecurrenceService; 17 test cases covering edge cases including one-time bills (frequency: once), overdue bill backlog processing, transaction failures, and multiple bill batching
- `lib/services/SchedulerService.ts` (174 lines) - Job registration timing (00:05 after daily-bill-check at 00:00) and dynamic import pattern to prevent Edge runtime bundling of node-cron
- `instrumentation.ts` (29 lines) - Runtime guard logic to prevent Edge runtime initialization of Node.js-only cron library

### ⚠️ Risk Assessment

- **Breaking Changes:** No - all changes are additive and do not alter existing public APIs or exported entity signatures.

- **Migrations/State:** No database migrations required. Changes operate on existing schema assuming bills table contains isAutoPay, status, dueDate, and isArchived columns.

### CODE REVIEW

File: lib/services/AutoPayService.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ *
+ * Users can mark bills as "auto-pay" to indicate their bank handles
+ * payment automatically (direct debit, recurring card charge, etc.).
+ * This service acknowledges those payments by:
+ * 1. Creating a transaction record (audit trail)
+ * 2. Advancing the bill to its next due date
+ *
+ * This preserves the "Active Payer" philosophy because the user
+ * consciously enabled auto-pay—it's a deliberate delegation.
+ */
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply lib/services/AutoPayService.ts]: This is an exception. In fact, there is no automatic, actual payment happening here, but the payment is simply marked as paid.

**Comment by sergeyklay** [comment]
@coderabbitai resolve
